ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -429,10 +464,9 @@ private void read(ChannelHandlerContext ctx, DataStreamRequestByteBuf request) {
           } else {
             throw new IllegalStateException(this + ": Unexpected type " + request.getType() + ", request=" + request);
           }
+          buf.release();
```

Comment:
If we throw IllegalStateException at previous line, we have no chance to release buf here.

Reply:
If it throws IllegalStateException, it is bug that should be fixed.  Just like assertion, we won't expect the code will work correctly if an assertion fails.